tests/ContextIntegrationTests.js

```diff
@@ -0,0 +1,323 @@
+                }, {
+                    func: "{logoutRequest}.send"
+                }, {
+                    event: "{logoutRequest}.events.onComplete",
+                    listener: "gpii.test.logoutRequestListen"
+                }
+            ],
+            gpii.test.createProcessChecks(gpii.tests.contextIntegration.data.processes, "expectRestored"),
+            [
+                {
+                    func: "gpii.test.checkRestoredConfiguration",
+                    args: ["{tests}.gpii-default.settingsHandlers", "{tests}.settingsStore", "{nameResolver}"]
+                }
+            ]
+        ]
+    }, {
+        name: "Multiple context changes",
+        expect: 9,
+        sequenceSegments: [
+            [
+                {
+                    "func": "gpii.test.expandSettings",
+                    args: [ "{tests}", [ "bright", "gpii-default" ]]
+                }, {
+                    func: "gpii.test.snapshotSettings",
+                    args: ["{tests}.gpii-default.settingsHandlers", "{tests}.settingsStore", "{nameResolver}"]
+                }, {
+                    func: "{loginRequest}.send"
+                }, {
+                    event: "{loginRequest}.events.onComplete",
+                    listener: "gpii.test.loginRequestListen"
+                }, {
+                    func: "gpii.test.checkConfiguration",
+                    args: ["{tests}.gpii-default.settingsHandlers", "{nameResolver}"]
+                }
+            ],
+            gpii.test.createProcessChecks(gpii.tests.contextIntegration.data.processes, "expectConfigured"),
+            gpii.tests.contextIntegration.changeEnvironmentAndCheck("bright"),
+            gpii.tests.contextIntegration.changeEnvironmentAndCheck("noise"),
+            gpii.tests.contextIntegration.changeEnvironmentAndCheck("brightandnoise"),
+
+            [
+                {
+                    func: "{logoutRequest}.send"
+                }, {
+                    event: "{logoutRequest}.events.onComplete",
+                    listener: "gpii.test.logoutRequestListen"
+                }
+            ],
+            gpii.test.createProcessChecks(gpii.tests.contextIntegration.data.processes, "expectRestored"),
+            [
+                {
+                    func: "gpii.test.checkRestoredConfiguration",
+                    args: ["{tests}.gpii-default.settingsHandlers", "{tests}.settingsStore", "{nameResolver}"]
+                }
+            ]
+        ]
+    }
+];
+
+gpii.tests.contextIntegration.buildTestFixtures = function (fixtures) {
+    return fluid.transform(fixtures, function (fixture) {
+        var testDef = {
+            name: fixture.name,
+            userToken: "context1",
+            expect: fixture.expect,
+            gradeNames: "gpii.tests.contextIntegration.testCaseHolder.linux",
+            config: {
+                configName: "linux-builtIn-config",
+                configPath: "tests/platform/linux/configs"
```

Comment:
These tests fail for me when run from their own directory (or indeed from all-tests.js) I believe because of this hard-coded relative path. The failure is

Error: ENOENT, no such file or directory 'E:\source\gits\gpii\node_modules\universal\tests\tests\platform\linux\configs\linux-builtIn-config.json'

I think that relative paths by default are interpreted relative to the process working directory. You should follow the pattern in DevelopmentTests.js, annoying though it is, and explicitly resolve a path with respect to __dirname which is guaranteed to be the directory holding the module file itself.